### PR TITLE
REACT1001 - [2] - add config and code changes

### PR DESCRIPTION
**Pull request by fcbjimmy**
1. Installed eslint
2. added prettier config file
3. made some code changes

### CODE REVIEW

File: src/components/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            Sign in
+          </Button>
+          <Button
+            type='submit'
+            color='primary'
+            fullWidth
+            variant='contained'
+            style={btnStyle}
+            onClick={signInWithGmail}
+            sx={{ display: 'flex', justifyContent: 'center' }}
```
Comment (cheungjaidev) [inline src/components/Login.jsx:85]: Is this inline setting from material-ui?

File: src/components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          >
+            <GoogleIcon sx={{ marginRight: '2px' }} fontSize='small' />
```
Comment (cheungjaidev) [inline src/components/Login.jsx:87]: Is this inline setting from material-ui?

File: src/containers/Index.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        </Box>
+      )}
+    </div>
+  );
+}
+
+function SignInOutContainer() {
+  const [value, setValue] = useState(0);
+
+  const handleChange = (event, newValue) => {
```
Comment (cheungjaidev) [inline src/containers/Index.jsx]: Will this handleChange function be reuseable?

File: package.json [hunk 4]
```diff
@@ -44,6 +44,16 @@
     ]
   },
   "devDependencies": {
-    "gh-pages": "^3.2.3"
+    "babel-eslint": "^10.1.0",
```
Comment (yutooo5225) [inline package.json]: npm **WARN** deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.

https://www.npmjs.com/package/@babel/eslint-parser

File: src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  const { loading, error, user } = useUserContext();
+
+  const test = user ? <Dashboard /> : <SignInOutContainer />;
+
+  return (
+    <>
+      {error && <p>{error}</p>}
+      {loading ? <h2>Loading...</h2> : test}
+    </>
+  );
```
Comment (yutooo5225) [inline src/App.jsx]: I think remove `Line 10` and change `Line 15`  to below style will be better.
```
{loading && <h2>Loading...</h2>}
{!loading && user && <Dashboard />}
{!loading && user == null && <SignInOutContainer />}

File: src/components/Signup.jsx [hunk 6]
```diff
@@ -1,31 +1,24 @@
...
+    padding: '20px',
+    width: '300px',
+    height: '60vh',
+    margin: '0 auto',
   };
   const headerStyle = { margin: 0 };
-  const avatarStyle = { backgroundColor: "#1bbd7e" };
-  const btnStyle = { margin: "20px 0" };
+  const avatarStyle = { backgroundColor: '#1bbd7e' };
+  const btnStyle = { margin: '20px 0' };
```
Comment (yutooo5225) [inline src/components/Signup.jsx]: Should be outside the `Signup` function

File: src/components/Login.jsx (hunk 1 above)
Comment (fcbjimmy) [reply src/components/Login.jsx:85]: Yes, The sx prop is a shortcut for defining custom style that has access to the Material UI theme. So we can change the colour of the theme directly from the material ui theme with the sx prop.

File: src/components/Login.jsx (hunk 2 above)
Comment (fcbjimmy) [reply src/components/Login.jsx:87]: Yes, The sx prop is a shortcut for defining custom style that has access to the Material UI theme. So we can change the colour of the theme directly from the material ui theme with the sx prop.

**Review by yutooo5225 — APPROVED** [review]

**Review by yutooo5225 — COMMENTED** [review]
1. If the value changed *frequently* in state, it's better to consider to use *debounce* before update state.
2. for those function provided by `firebase/auth`, better to use `async await` to impove the code readability

**Pull request merged**